wgengine/magicsock/magicsock.go

```diff
@@ -1051,6 +1075,22 @@ func (c *Conn) determineEndpoints(ctx context.Context) ([]tailcfg.Endpoint, erro
 		addAddr(ipp(localAddr.String()), tailcfg.EndpointLocal)
 	}
 
+	select {
+	case ext, ok := <-portmap:
+		if ok {
+			addAddr(ext, tailcfg.EndpointPortmapped)
+			c.setNetInfoHavePortMap()
+		}
+	default:
+		// fire off a goroutine which will update netinfo later
+		go func() {
+			if ext, ok := <-portmap; ok {
+				c.setNetInfoHavePortMap()
```

Comment:
Hm, I think both of these lines can trigger MapRequest updates. Ideally they'd ride along together to the control server.

Reply:
Wouldn't this have been happening previously as well? I'm not sure I'm able to change that at this level since they're both behind function pointers, although I could add an extra flag to only invoke it once or something?